androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java

```diff
@@ -0,0 +1,1048 @@
+        onView(withId(R.id.shareViaShowFileListingSwitch)).check(matches(not(isDisplayed())));
+
+        //Setting a password... no matter which one
+        onView(withId(R.id.shareViaLinkPasswordSwitch)).perform(click());
+        onView(withId(R.id.shareViaLinkPasswordValue)).perform(scrollTo(), replaceText("a"));
+
+        //Enable "expiration"
+        onView(withId(R.id.shareViaLinkExpirationSwitch)).perform(click());
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+        onView(withId(android.R.id.button1)).perform(click());
+        onView(withId(R.id.confirmAddPublicLinkButton)).perform(scrollTo(), click());
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+
+        //Skip the sharing panel
+        pressBack();
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+
+        onView(withId(R.id.editPublicLinkButton)).perform(click());
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+
+        //Check the status of the sharing options
+        onView(withId(R.id.shareViaLinkPasswordSwitch)).check(matches(isChecked()));
+        onView(withId(R.id.shareViaLinkExpirationSwitch)).check(matches(isChecked()));
+
+        onView(withId(R.id.cancelAddPublicLinkButton)).perform(scrollTo(), click());
+
+        Log_OC.i(LOG_TAG, "Test Enable Edit Options File Passed");
+
+    }
+
+    /**
+     *  TEST CASE: Remove public link on a file
+     *  PASSED IF: No links in share view
+     */
+    @Test
+    public void test_18_unshare_public_file()
+            throws InterruptedException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+
+        Log_OC.i(LOG_TAG, "Test Unshare Public File Start");
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+
+        //Select share option
+        selectShare(file);
+
+        //Check that a public link exists
+        onView(withId(R.id.shareNoPublicLinks))
+                .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.GONE)));
+
+        //Delete link
+        onView(withId(R.id.deletePublicLinkButton)).perform(click());
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+        onView(withId(android.R.id.button1)).perform(click());
+        SystemClock.sleep(WAIT_CONNECTION_MS);
+
+        //Check that there are no public links
+        onView(withId(R.id.shareNoPublicLinks)).check(matches(isDisplayed()));
+
+        Log_OC.i(LOG_TAG, "Test Unshare Public File Passed");
+
+    }
+
+
+    //To create a new public link with defaults
+    private void publicShareCreationDefault (String name) {
+
+        //Creation of the share link. Name only for servers >= 10
+        onView(withId(R.id.addPublicLinkButton)).perform(click());
+
+        //Check server version and parameter null (or not) to handle the link name
+        if (capabilities.getVersionMayor() >= VERSION_10 && name!=null) {
+            onView(withId(R.id.shareViaLinkNameValue)).perform(replaceText(name));
+        }
+
+        SystemClock.sleep(WAIT_CONNECTION_MS);
```

Comment:
Are all the waits necessary? I've read something about Espresso now waiting for the views to be ready, or something similar. Talking from memory, not something I really know...